stack_data/formatting.py

```diff
@@ -184,14 +195,28 @@ def format_line(self, line: Line) -> str:
             for line_range in line.executing_node_ranges:
                 start = line_range.start - line.leading_indent
                 end = line_range.end - line.leading_indent
-                result += (
-                        " " * (start + len(prefix))
-                        + self.executing_node_underline * (end - start)
-                        + "\n"
-                )
-
+                # if end <= start, we have an empty line inside a highlighted
+                # block of code. In this case, we need to avoid inserting
+                # an extra blank line with no markers present.
+                if end > start:
+                    result += (
+                            " " * (start + len(prefix))
+                            + self.executing_node_underline * (end - start)
+                            + "\n"
+                    )
         return result
 
+
+    def format_blank_lines_linenumbers(self, blank_line):
+        if self.current_line_indicator:
+            result = " " * len(self.current_line_indicator)
+        else:
+            result = "   "
```

Comment:
https://coveralls.io/builds/51814297/source?filename=stack_data%2Fformatting.py#L214 points out that this line is uncovered, and it seems worth testing.

Reply:
Ok. I probably won't have time to do this tonight though.

Reply:
I had added this `else` part based on the presence of the second term in https://github.com/alexmojaki/stack_data/blob/7e8808252fde3fbd5e5fbd84ae2913e98cc77fde/stack_data/formatting.py#L173
However, it turns out that `result` was never null and that this line was redundant in the original code, based on all existing tests. So I removed the corresponding `else` part of my addition.

Reply:
It's pretty easy to see when the line isn't redundant in the previous code: when you set `show_linenos=False, current_line_indicator=""`. `result` won't be null, it'll be an empty string. Defaulting it to spaces ensures that the user's code is indented like in a standard traceback. I guess it should have been tested before, but coverage didn't see the gap.

Reply:
Do you want me to reinstate it and write a relevant test for it?

Reply:
Yes please.

Reply:
Looks like there's been some confusion or misremembering. The test needs to specifically have both the arguments `show_linenos=False, current_line_indicator=""` in one formatter.

Reply:
Yes, I was wondering why the line `result = result or "   "` did not seem to have any effect. I will fix this.